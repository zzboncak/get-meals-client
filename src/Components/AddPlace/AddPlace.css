.add-location-form {
    background: #fa9843;
    width: 75%;
    padding: 40px;
    margin: 0 auto;
    border-radius: 8px;
    box-shadow: 5px 5px 5px 5px rgba(0, 0, 0, 0.5);
    border: 2px solid #ce7a30;
    justify-content: center;
    display: flex;
}

.add-location {
    width: 95%;
    margin-left: auto;
    margin-right: auto;
    font-family: 'Ubuntu';
    padding-top: 25px;
}

form {
    margin: 20px 100px;
    padding: 20px 75px;
    text-align: center;
}

label {
    font-weight: bold;
    display: flex;
    flex-direction: column;
}

.place-form__label {
    display: flex;
    flex-direction: row;
    text-align: left;
    /* padding: 0px 0px; */
    margin: 0px 64px;
    flex: 1 1;
}

.place-form__input {
    display: flex;
    flex-direction: row;
    text-align: right;
    flex: 1 1;
}

.place-form__inputs {
    display: flex;
    flex-direction: row;
    flex: 1 1;
}



span {
    color: #af0303
}

.place-form__days {
    display: flex;
    flex-direction: row;
    text-align: left;
    /* padding: 0px 0px; */
    margin: 0px 64px;
    flex: 1 1;
}

.place-form__select {
    border: 1px solid whitesmoke;
    width: 120px;
    border-radius: 3px;
    overflow: hidden;
    background: whitesmoke;
    justify-content: center;
}

.place-form__select select {
    padding: 5px 8px;
    width: 130%;
    border: none;
    box-shadow: none;
    background: transparent;
    background-image: none;
    -webkit-appearance: none;
}

.place-form__select select:focus {
    outline: none;
}

button {
    margin-top:10px;
    height: auto;
    font-size: 20px;
    font-weight: 500;
    text-align: center;
    color:black;
    background: #fd7702;
    border: 0;
    cursor: pointer;
    margin-bottom: 0;
    border-radius:3px;
    padding: 10px 20px;
    box-shadow: 2px 2px 2px 2px rgba(0, 0, 0, 0.5);
  }

button:hover,
button:active {
    color: #fd7702;
    background: #7a3700;
}

.add-location-form-error-message {
    color: red;
}

@media screen and (min-width: 320px) {
    .place-form__days {
        display: flex;
        flex-direction: column;
        flex: 1 1;
    }
}

@media screen and (min-width: 1600px) {
    .place-form__days {
        display: flex;
        flex-direction: row;
        flex: 1 1;
    }
}