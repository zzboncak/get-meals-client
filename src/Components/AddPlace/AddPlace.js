import React from 'react';
import './AddPlace.css';
import GetMealsContext from '../../GetMealsContext';

class AddPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: {
                value: '',
                isTouched: false
            },
            placeAddress: {
                value: '',
                isTouched: false
            },
            city: {
                value: '',
                isTouched: false
            },
            usState: {
                value: '',
                isTouched: false
            },
            zipcode: {
                value: '',
                isTouched: false
            },
            website: {
                value: '',
                isTouched: false
            },
            hourOfOperation: {
                value: '',
                isTouched: false
            },
            dateOfOperation: {
                value: '',
                isTouched: false
            },
            typeOfFood: {
                value: '',
                isTouched: false
            },
        }
    }

    static contextType = GetMealsContext;

    generatePlaceId = () => {
        let placeId = Math.ceil(Math.random()*1000000);
        return placeId;
    }

    handleSubmitAddPlace = (e) => {
        e.preventDefault();
        let newLocation = {
            location_name: this.state.name,
            street_address: this.state.placeAddress,
            city: this.state.city,
            state: this.state.usState,
            zip: this.state.zipcode
        };

        this.context.getGooglePlaceID(this.state.placeAddress + ' ' + this.state.city + ' ' + this.state.usState + ' ' + this.state.zipcode)

        const url = `https://frozen-everglades-23155.herokuapp.com/api/locations`;

        const options = {
            method: 'POST',
            body: JSON.stringify(newLocation),
            headers: {
                "Content-Type": "application/json"
            }
        };

        fetch(url, options)
            .then(res => {
                if(!res.ok) {
                    throw new Error('Something went wrong, please try again later');
                }
                return res.json();
            })
            .then(data => {
                this.setState({
                    name: {
                        value: '',
                        isTouched: false
                    },
                    placeAddress: {
                        value: '',
                        isTouched: false
                    },
                    city: {
                        value: '',
                        isTouched: false
                    },
                    usState: {
                        value: '',
                        isTouched: false
                    },
                    zipcode: {
                        value: '',
                        isTouched: false
                    },
                    website: {
                        value:'',
                        isTouched: false
                    },
                    hourOfOperation: {
                        value: '',
                        isTouched: false
                    },
                    dateOfOperation: {
                        value: '',
                        isTouched: false
                    },
                    typeOfFood: {
                        value: '',
                        isTouched: false
                    },
                });
                this.context.locationFetch();
                this.props.history.push('/');
            })
            .catch(err => {
                console.log(err)
            });
    }
    
    onNameChange = (newName) => {
        this.setState({
            name: newName
        });
    }

    onAddressChange = (newAddress) => {
        this.setState({
            placeAddress: newAddress
        });
    }

    onCityChange = (newCity) => {
        this.setState({
            city: newCity
        });
    }

    onUsStateChange = (newState) => {
        this.setState({
            usState: newState
        });
    }


    onWebsiteChange = (newWebsite) => {
        this.setState({
            website: newWebsite
        });
    }

    onZipcodeChange = (newZipcode) => {
        this.setState({
            zipcode: newZipcode
        });
    }

    onHoursChange = (newHours) => {
        this.setState({
            hourOfOperation: newHours
        });
    }

    onDateChange = (newDate) => {
        this.setState({
            dateOfOperation: newDate
        });
    }

    onTypeChange = (newFood) => {
        this.setState({
            typeOfFood: newFood
        });
    }

    validatePlaceName() {
        if(this.state.name.length < 1){
            return 'You must enter a location name';
        }
    }

    validatePlaceAddress() {
        if (this.state.placeAddress === '') {
            return 'Your address needs content';
        }
    }

    validateCity() {
        if (this.state.city === '') {
            return 'You need to enter a City'
        }
    }

    validateUsState() {
        if (this.state.usState === '') {
            return 'You need to enter a State'
        }
    }

    validateWebsite() {
        if (this.state.website === '') {
            return 'You need to enter a website'
        }
    }

    validateZipcode() {
        if (this.state.zipcode === '') {
            return 'You need to enter a zipcode'
        }
    }

    validateHoursOfOperation() {
        if (this.state.hourOfOperation === '') {
            return 'You must select hours of operation'
        }
    }

    validateDateOfOperation() {
        if (this.state.hourOfOperation === '') {
            return 'You must select date of operation'
        }
    }

    validateTypeOfFood() {
        if (this.state.typeOfFood === '') {
            return 'You must select location type'
        }
    }

    render() {

        const nameError = this.validatePlaceName();
        const addressError = this.validatePlaceAddress();
        const cityError = this.validateCity();
        const stateError = this.validateUsState();
        const zipcodeError = this.validateZipcode();
        const websiteError = this.validateWebsite()
        const hoursError = this.validateHoursOfOperation();
        const dateError = this.validateDateOfOperation();
        const typeError = this.validateTypeOfFood();

        return (
            <div>
                <div className='add-location'>
                    <h2>This is where you come in!</h2>
                    <h4>We ask that if you know of any locations that provide free meals to add the contact information so that other users can easily find food by just searching a city in the US</h4>
                </div>
            <div className='add-location-form'>
                <form onSubmit={e => this.handleSubmitAddPlace(e)} className='place-form__start'>
                    <div className='place-form__inputs'>
                        <label htmlFor='add-location' className='place-form__location-name place-form__label'>Location name: </label>
                        <span>*</span>
                        <input 
                            type='text' 
                            placeholder='name of location' 
                            name='add-location' 
                            value={this.state.name.value} 
                            className='place-form__input'
                            onChange={e => this.onNameChange(e.target.value)} 
                        />
                    </div>
                    <br />
                    {this.state.name.isTouched && nameError}
                    <br />

                    <div className='place-form__inputs'>
                        <label htmlFor='place-address' className='place-form__address place-form__label'>Address:<span>* </span></label>
                        <input 
                            name='place-address' 
                            id='place-address' 
                            value={this.state.placeAddress.value} 
                            className='place-form__input'
                            onChange={e => this.onAddressChange(e.target.value)} 
                        />
                    </div>
                    <br />
                    {this.state.placeAddress.isTouched && addressError}
                    <br />

                    <div className='place-form__inputs'>
                        <label htmlFor='city-location' className='place-form__city place-form__label'>City:<span>*</span> </label>
                        <input 
                            name='city-location' 
                            id='city-location' 
                            value={this.state.city.value} 
                            className='place-form__input'
                            onChange={e => this.onCityChange(e.target.value)} 
                        />
                    </div>
                    <br />
                    {this.state.city.isTouched && cityError}
                    <br />

                    <div className='place-form__inputs'>
                        <label htmlFor='us-state' className='place-form__state place-form__label'>State:<span>*</span> </label>
                        <input
                            name='us-state' 
                            id='us-state' 
                            value={this.state.usState.value} 
                            className='place-form__input'
                            onChange={e => this.onUsStateChange(e.target.value)} 
                        />
                    </div>
                    <br />
                    {this.state.usState.isTouched && stateError}
                    <br />

                    <div className='place-form__inputs'>
                        <label htmlFor='zipcode' className='place-form__state place-form__label'>Zipcode:<span>*</span> </label>
                        <input
                            type='number'
                            name='zipcode' 
                            id='zipcode' 
                            value={this.state.zipcode.value} 
                            className='place-form__input'
                            onChange={e => this.onZipcodeChange(e.target.value)} 
                        />
                    </div>
                    <br />
                    {this.state.zipcode.isTouched && zipcodeError}
                    <br />

                    <div className='place-form__inputs'>
                        <label htmlFor='website' className='place-form__website place-form__label'>Website: </label>
                        <input
                            name='website'
                            id='website'
                            value={this.state.website.value}
                            className='place-form__6input'
                            onChange={e => this.onWebsiteChane(e.target.value)}
                        />
                    </div>
                        <br />
                        <br />

                    <div className='place-form__inputs'>
                        <label htmlFor='date-of-operation' className='place-form__date-of-operation'>Days of Operation:</label>
                        <br /><br />
                        <div className='place-form__days'>
                            <input 
                                type='checkbox' 
                                name='date-check' 
                                value={this.state.dateOfOperation.value}
                                className='place-form__day-input'
                                onChange={e => this.onDateChange(e.target.value)}
                            />
                            <label htmlFor='date-check'>Monday</label>
                            <label>Open</label>
                            <input 
                                type='time' 
                                name='opening-time'
                                className='place-form__open-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <label>Close</label>
                            <input 
                                type='time' 
                                name='closing-time'
                                className='place-form__close-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <br />
                            {this.state.dateOfOperation.isTouched && stateError}
                            {this.state.hourOfOperation.isTouched && stateError}
                            <br />
                        </div>
                    </div>

                    <div className='place-form__inputs'>
                        <div className='place-form__days'>
                            <input 
                                type='checkbox' 
                                name='date-check' 
                                value={this.state.dateOfOperation.value}
                                className='place-form__day-input'
                                onChange={e => this.onDateChange(e.target.value)}
                            />
                            <label htmlFor='date-check'>Tuesday</label>
                            <label>Open</label>
                            <input 
                                type='time' 
                                name='opening-time'
                                className='place-form__open-input'
                                onChange={e => this.onDateChange(e.target.value)}
                                />
                            <label>Close</label>
                            <input 
                                type='time' 
                                name='closing-time'
                                className='place-form__close-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <br />
                            {this.state.dateOfOperation.isTouched && stateError}
                            {this.state.hourOfOperation.isTouched && stateError}
                            <br />
                        </div>
                    </div>  
                        
                    <div className='place-form__inputs'>
                        <div className='place-form__days'>
                            <input 
                                type='checkbox' 
                                name='date-check' 
                                value={this.state.dateOfOperation.value}
                                className='place-form__day-input'
                                onChange={e => this.onDateChange(e.target.value)}
                            />
                            <label htmlFor='date-check'>Wednesday</label>
                            <label>Open</label>
                            <input 
                                type='time' 
                                name='opening-time'
                                className='place-form__open-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <label>Close</label>
                            <input 
                                type='time' 
                                name='closing-time'
                                className='place-form__close-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <br />
                            {this.state.dateOfOperation.isTouched && stateError}
                            {this.state.hourOfOperation.isTouched && stateError}
                            <br />
                        </div>
                    </div>
                        
                    <div className='place-form__inputs'>
                        <div className='place-form__days'>
                            <input 
                                type='checkbox' 
                                name='date-check' 
                                value={this.state.dateOfOperation.value}
                                className='place-form__day-input'
                                onChange={e => this.onDateChange(e.target.value)}
                            />
                            <label htmlFor='date-check'>Thursday</label>
                            <label>Open</label>
                            <input 
                                type='time' 
                                name='opening-time'
                                className='place-form__open-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <label>Close</label>
                            <input 
                                type='time' 
                                name='closing-time'
                                className='place-form__close-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <br />
                            {this.state.dateOfOperation.isTouched && stateError}
                            {this.state.hourOfOperation.isTouched && stateError}
                            <br />
                        </div>
                    </div>
                        
                    <div className='place-form__inputs'>
                        <div className='place-form__days'>
                            <input 
                                type='checkbox' 
                                name='date-check' 
                                value={this.state.dateOfOperation.value}
                                className='place-form__day-input'
                                onChange={e => this.onDateChange(e.target.value)}
                            />
                            <label htmlFor='date-check'>Friday</label>
                            <label>Open</label>
                            <input 
                                type='time' 
                                name='opening-time'
                                className='place-form__open-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <label>Close</label>
                            <input 
                                type='time' 
                                name='closing-time'
                                className='place-form__close-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <br />
                            {this.state.dateOfOperation.isTouched && stateError}
                            {this.state.hourOfOperation.isTouched && stateError}
                            <br />
                        </div>
                    </div>

                    <div className='place-form__inputs'>
                        <div className='place-form__days'>
                            <input 
                                type='checkbox' 
                                name='date-check' 
                                value={this.state.dateOfOperation.value}
                                className='place-form__day-input'
                                onChange={e => this.onDateChange(e.target.value)}
                            />
                            <label htmlFor='date-check'>Saturday</label>
                            <label>Open</label>
                            <input 
                                type='time' 
                                name='opening-time'
                                className='place-form__open-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <label>Close</label>
                            <input 
                                type='time' 
                                name='closing-time'
                                className='place-form__close-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <br />
                            {this.state.dateOfOperation.isTouched && stateError}
                            {this.state.hourOfOperation.isTouched && stateError}
                            <br />
                        </div>
                    </div>
                        
                    <div className='place-form__inputs'>
                        <div className='place-form__days'>
                            <input 
                                type='checkbox' 
                                name='date-check' 
                                value={this.state.dateOfOperation.value}
                                className='place-form__day-input'
                                onChange={e => this.onDateChange(e.target.value)}
                            />
                            <label htmlFor='date-check'>Sunday</label>
                            <label>Open</label>
                            <input 
                                type='time' 
                                name='opening-time'
                                className='place-form__open-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <label>Close</label>
                            <input 
                                type='time' 
                                name='closing-time'
                                className='place-form__close-input'
                                onChange={e => this.onDateChange(e.target.value)}/>
                            <br />
                            {this.state.dateOfOperation.isTouched && stateError}
                            {this.state.hourOfOperation.isTouched && stateError}
                            <br />
                        </div>
                    </div>

                    <div className='place-form__inputs'>
                        <label htmlFor='location-type' className='place-form__location-types'>Location Type:  </label>
                        <span>*  </span> 
                        <div className='place-form__select'>
                            <select>
                                <option className='place-form__location-type'>Restaurant</option>
                                <option className='place-form__location-type'>Food Bank</option>
                                <option className='place-form__location-type'>School</option>
                                <option className='place-form__location-type'>Other Non-Profit</option>
                            </select>
                        </div>
                    </div>
                        <br />
                        {this.state.typeOfFood.isTouched && typeError}
                        <br />
                        
                        <button onClick={this.context.addNewPlace}
                            type='submit' 
                            className='place-form__button-submit'
                            disabled={
                                this.validatePlaceName() ||
                                this.validatePlaceAddress() ||
                                this.validateCity() ||
                                this.validateUsState() ||
                                this.validateWebsite() ||
                                this.validateHoursOfOperation() ||
                                this.validateDateOfOperation() ||
                                this.validateTypeOfFood() 
                        }>
                            Add this location
                        </button>
                    </form>
                </div>
            </div>
        );
    }
}


    

export default AddPage