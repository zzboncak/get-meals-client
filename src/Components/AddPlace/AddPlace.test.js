import React from 'react';
import ReactDOM from 'react-dom';
import AddPlace from './AddPlace';

describe(`AddPlace component`, () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<AddPlace />, div);
      ReactDOM.unmountComponentAtNode(div);
  });
})
