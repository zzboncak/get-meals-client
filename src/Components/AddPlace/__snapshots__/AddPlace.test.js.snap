// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlace component renders the component as expected 1`] = `
<div>
  <h2>
    This is where you come in!
  </h2>
  <h4>
    We ask that if you know of any locations that provide free meals to add the contact information so that other users can easily find food by just searching a city in the US
  </h4>
  <div
    className="add-location-form"
  >
    <form
      className="place-form__start"
      onSubmit={[Function]}
    >
      <div
        className="place-form__inputs"
      >
        <label
          className="place-form__location-name place-form__label"
          htmlFor="add-location"
        >
          Location name: 
        </label>
        <span>
          *
        </span>
        <input
          className="place-form__input"
          name="add-location"
          onChange={[Function]}
          placeholder="name of location"
          type="text"
          value=""
        />
      </div>
      <br />
      <br />
      <div
        className="place-form__inputs"
      >
        <label
          className="place-form__address place-form__label"
          htmlFor="place-address"
        >
          Address:
          <span>
            * 
          </span>
        </label>
        <input
          className="place-form__input"
          id="place-address"
          name="place-address"
          onChange={[Function]}
          value=""
        />
      </div>
      <br />
      <br />
      <div
        className="place-form__inputs"
      >
        <label
          className="place-form__city place-form__label"
          htmlFor="city-location"
        >
          City:
          <span>
            *
          </span>
           
        </label>
        <input
          className="place-form__input"
          id="city-location"
          name="city-location"
          onChange={[Function]}
          value=""
        />
      </div>
      <br />
      <br />
      <div
        className="place-form__inputs"
      >
        <label
          className="place-form__state place-form__label"
          htmlFor="us-state"
        >
          State:
          <span>
            *
          </span>
           
        </label>
        <input
          className="place-form__input"
          id="us-state"
          name="us-state"
          onChange={[Function]}
          value=""
        />
      </div>
      <br />
      <br />
      <div
        className="place-form__inputs"
      >
        <label
          className="place-form__state place-form__label"
          htmlFor="zipcode"
        >
          Zipcode:
          <span>
            *
          </span>
           
        </label>
        <input
          className="place-form__input"
          id="zipcode"
          name="zipcode"
          onChange={[Function]}
          type="number"
          value=""
        />
      </div>
      <br />
      <br />
      <div
        className="place-form__inputs"
      >
        <label
          className="place-form__website place-form__label"
          htmlFor="website"
        >
          Website: 
        </label>
        <input
          className="place-form__6input"
          id="website"
          name="website"
          onChange={[Function]}
          value=""
        />
      </div>
      <br />
      <br />
      <div
        className="place-form__inputs"
      >
        <label
          className="place-form__date-of-operation"
          htmlFor="date-of-operation"
        >
          Days of Operation:
        </label>
        <br />
        <br />
        <div
          className="place-form__days"
        >
          <input
            className="place-form__day-input"
            name="date-check"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          <label
            htmlFor="date-check"
          >
            Monday
          </label>
          <label>
            Open
          </label>
          <input
            className="place-form__open-input"
            name="opening-time"
            onChange={[Function]}
            type="time"
          />
          <label>
            Close
          </label>
          <input
            className="place-form__close-input"
            name="closing-time"
            onChange={[Function]}
            type="time"
          />
          <br />
          <br />
        </div>
      </div>
      <div
        className="place-form__inputs"
      >
        <div
          className="place-form__days"
        >
          <input
            className="place-form__day-input"
            name="date-check"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          <label
            htmlFor="date-check"
          >
            Tuesday
          </label>
          <label>
            Open
          </label>
          <input
            className="place-form__open-input"
            name="opening-time"
            onChange={[Function]}
            type="time"
          />
          <label>
            Close
          </label>
          <input
            className="place-form__close-input"
            name="closing-time"
            onChange={[Function]}
            type="time"
          />
          <br />
          <br />
        </div>
      </div>
      <div
        className="place-form__inputs"
      >
        <div
          className="place-form__days"
        >
          <input
            className="place-form__day-input"
            name="date-check"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          <label
            htmlFor="date-check"
          >
            Wednesday
          </label>
          <label>
            Open
          </label>
          <input
            className="place-form__open-input"
            name="opening-time"
            onChange={[Function]}
            type="time"
          />
          <label>
            Close
          </label>
          <input
            className="place-form__close-input"
            name="closing-time"
            onChange={[Function]}
            type="time"
          />
          <br />
          <br />
        </div>
      </div>
      <div
        className="place-form__inputs"
      >
        <div
          className="place-form__days"
        >
          <input
            className="place-form__day-input"
            name="date-check"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          <label
            htmlFor="date-check"
          >
            Thursday
          </label>
          <label>
            Open
          </label>
          <input
            className="place-form__open-input"
            name="opening-time"
            onChange={[Function]}
            type="time"
          />
          <label>
            Close
          </label>
          <input
            className="place-form__close-input"
            name="closing-time"
            onChange={[Function]}
            type="time"
          />
          <br />
          <br />
        </div>
      </div>
      <div
        className="place-form__inputs"
      >
        <div
          className="place-form__days"
        >
          <input
            className="place-form__day-input"
            name="date-check"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          <label
            htmlFor="date-check"
          >
            Friday
          </label>
          <label>
            Open
          </label>
          <input
            className="place-form__open-input"
            name="opening-time"
            onChange={[Function]}
            type="time"
          />
          <label>
            Close
          </label>
          <input
            className="place-form__close-input"
            name="closing-time"
            onChange={[Function]}
            type="time"
          />
          <br />
          <br />
        </div>
      </div>
      <div
        className="place-form__inputs"
      >
        <div
          className="place-form__days"
        >
          <input
            className="place-form__day-input"
            name="date-check"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          <label
            htmlFor="date-check"
          >
            Saturday
          </label>
          <label>
            Open
          </label>
          <input
            className="place-form__open-input"
            name="opening-time"
            onChange={[Function]}
            type="time"
          />
          <label>
            Close
          </label>
          <input
            className="place-form__close-input"
            name="closing-time"
            onChange={[Function]}
            type="time"
          />
          <br />
          <br />
        </div>
      </div>
      <div
        className="place-form__inputs"
      >
        <div
          className="place-form__days"
        >
          <input
            className="place-form__day-input"
            name="date-check"
            onChange={[Function]}
            type="checkbox"
            value=""
          />
          <label
            htmlFor="date-check"
          >
            Sunday
          </label>
          <label>
            Open
          </label>
          <input
            className="place-form__open-input"
            name="opening-time"
            onChange={[Function]}
            type="time"
          />
          <label>
            Close
          </label>
          <input
            className="place-form__close-input"
            name="closing-time"
            onChange={[Function]}
            type="time"
          />
          <br />
          <br />
        </div>
      </div>
      <div
        className="place-form__inputs"
      >
        <label
          className="place-form__location-types"
          htmlFor="location-type"
        >
          Location Type:  
        </label>
        <span>
          *  
        </span>
        <div
          className="place-form__select"
        >
          <select>
            <option
              className="place-form__location-type"
            >
              Restaurant
            </option>
            <option
              className="place-form__location-type"
            >
              Food Bank
            </option>
            <option
              className="place-form__location-type"
            >
              School
            </option>
            <option
              className="place-form__location-type"
            >
              Other Non-Profit
            </option>
          </select>
        </div>
      </div>
      <br />
      <br />
      <button
        className="place-form__button-submit"
        type="submit"
      >
        Add this location
      </button>
    </form>
  </div>
</div>
`;
